test/Lifecycle-test.ts

```diff
@@ -628,16 +647,132 @@ describe("Lifecycle", () => {
+                        accessToken,
+                        homeserverUrl,
+                        identityServerUrl,
+                        deviceId,
+                        freshLogin: true,
+                        guest: false,
+                        pickleKey: expect.any(String),
+                    },
+                    undefined,
+                );
+            });
+        });
+
+        describe("when authenticated via OIDC native flow", () => {
+            const clientId = "test-client-id";
+            const issuer = "https://auth.com/";
+
+            const delegatedAuthConfig = makeDelegatedAuthConfig(issuer);
+            const idTokenClaims = {
+                aud: "123",
+                iss: issuer,
+                sub: "123",
+                exp: 123,
+                iat: 456,
+            };
+
+            beforeAll(() => {
+                fetchMock.get(
+                    `${delegatedAuthConfig.issuer}.well-known/openid-configuration`,
+                    delegatedAuthConfig.metadata,
+                );
+                fetchMock.get(`${delegatedAuthConfig.issuer}jwks`, {
+                    status: 200,
+                    headers: {
+                        "Content-Type": "application/json",
+                    },
+                    keys: [],
+                });
+            });
+
+            beforeEach(() => {
+                // mock oidc config for oidc client initialisation
+                mockClient.getClientWellKnown.mockReturnValue({
+                    [MatrixJs.M_AUTHENTICATION.stable!]: {
+                        issuer: issuer,
+                    },
                 });
+                initSessionStorageMock();
+                // set values in session storage as they would be after a successful oidc authentication
+                persistOidcAuthenticatedSettings(clientId, issuer, idTokenClaims);
+            });
+
+            it("should not try to create a token refresher without a refresh token", async () => {
+                await setLoggedIn(credentials);
+
+                // didn't try to initialise token refresher
+                expect(fetchMock).not.toHaveFetched(`${delegatedAuthConfig.issuer}.well-known/openid-configuration`);
+            });
+
+            it("should not try to create a token refresher without a deviceId", async () => {
+                await setLoggedIn({
+                    ...credentials,
+                    refreshToken,
+                    deviceId: undefined,
+                });
+
+                // didn't try to initialise token refresher
+                expect(fetchMock).not.toHaveFetched(`${delegatedAuthConfig.issuer}.well-known/openid-configuration`);
+            });
+
+            it("should not try to create a token refresher without an issuer in session storage", async () => {
+                persistOidcAuthenticatedSettings(
+                    clientId,
+                    // @ts-ignore set undefined issuer
```

Comment:
I don't understand what "set undefined later" means here. It looks like it is set to undefined right here?

Could you clarify, please?

Reply:
oh I should learn to read apparently